include/ck/tensor_operation/gpu/grid/gridwise_layernorm.hpp

```diff
@@ -0,0 +1,368 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2022, Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include "ck/utility/data_type.hpp"
+#include "ck/utility/reduction_common.hpp"
+#include "ck/utility/reduction_operator.hpp"
+#include "ck/utility/reduction_functions_accumulate.hpp"
+#include "ck/tensor_operation/gpu/block/reduction_functions_blockwise.hpp"
+#include "ck/tensor_operation/gpu/thread/reduction_functions_threadwise.hpp"
+#include "ck/tensor_operation/gpu/thread/threadwise_tensor_slice_transfer.hpp"
+#include "ck/tensor_operation/gpu/element/element_wise_operation.hpp"
+
+namespace ck {
+
+template <typename GridwiseReduction,
+          typename XDataType,
+          typename GammaDataType,
+          typename BetaDataType,
+          typename YDataType,
+          typename AccDataType,
+          typename GridDesc_M_K>
+__global__ void kernel_layernorm(const GridDesc_M_K in_grid_desc_m_k,
+                                 const GridDesc_M_K gamma_grid_desc_m_k,
```

Comment:
Since gamma and beta are 1D unit-stride vectors, can we use `make_insert_transform` for 2D broadcasting like the following kernel does? Doing so can avoid extra 2D address calculation. It is also very possible to save a lot of VGPRs too because this is essentially duplicating rows of gamma/beta vectors.

https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/9e4429f9c3c6c08da06a65cc880b094850c4cb4e/include/ck/tensor_operation/gpu/grid/gridwise_gemm_xdl_layernorm_cshuffle_v1.hpp#L685-L688

Reply:
@asroy   Should we assume dimension of gamma and beta equal to the dimension excluding batch, or keeping flexbile, let it be runtime parameter by user
PS: dimension excluding batch. it means 
eg1: NCHW -> calculate mean and var accross CHW (N dimension). gamma and beta is CHW (3) dimension 
eg2: in the case of transformer, assume m (in m*n) is the batch size. gamma and beta is n dimension here.

Reply:
Finally, I keep grid descriptor of gamma and beta be 1D (K) directly.
VGPR also be 1D (K).
Without any broadcast tricks.
6ed9ab3a55eed3dbf3c339227562391ebae01f5d